geode-core/src/main/java/org/apache/geode/internal/cache/TXCommitMessage.java

```diff
@@ -733,7 +733,9 @@ void firePendingCallbacks(List<EntryEventImpl> callbacks) {
     try {
       lastTransactionEvent = getLastTransactionEvent(callbacks);
     } catch (ServiceConfigurationError ex) {
-      logger.error(ex.getMessage());
+      if (!ex.getMessage().startsWith(TXLastEventInTransactionUtils.NO_INFORMATION_FOR_SENDER_ID)) {
+        logger.error(ex.getMessage());
+      }
```

Comment:
If this is not necessarily a configuration error, instead of removing the error log, I would update `TXLastEventInTransactionUtils.checkNoSendersGroupTransactionEvents()` to return `false` in case a sender is null as follows:

```
      if (sender == null || sender.mustGroupTransactionEvents()) {
        return false;
      }

```
A similar change would have to be done in `TXLastEventInTransactionUtils.doesSenderGroupTransactionEvents`:

`return sender == null || sender.mustGroupTransactionEvents();`

Reply:
When sender == null, there is no need to get the last transaction event in the node. There is no sender.

Reply:
There could be several senders for the same region configured in different servers.

If you opt for returning null in `getLastTransactionEvent` when one of the senders is not configured, I suggest, you 
also change `TXCommitMessage` and `TXState` so that when `getLastTransactionEvent` returns null, it considers that all events for the transaction are the last event for the transaction:
`boolean isLastTransactionEvent = isConfigError || lastTransactionEvent == null || ee.equals(lastTransactionEvent);`

The reason for it is that if another sender is configured and has group-transaction-events set to true, it will not try to get the last event from each transaction in vain delaying unnecessarily the delivery of batches.

Reply:
I have modified the code so that if the above mentioned situation will set isConfigError to be true.
You are right, lastTransactionEvent == null should be included as well.